components/filterModal/detailFilterModal.tsx

```diff
@@ -0,0 +1,175 @@
+'use client';
+
+import { ChangeEvent, useState } from 'react';
+import DatePicker from 'react-datepicker';
+
+import Image from 'next/image';
+
+import 'react-datepicker/dist/react-datepicker.css';
+import Button from '../button/button';
+
+const ADDRESS_LIST = [
+  '서울시 종로구',
+  '서울시 중구',
+  '서울시 용산구',
+  '서울시 성동구',
+  '서울시 광진구',
+  '서울시 동대문구',
+  '서울시 중랑구',
+  '서울시 성북구',
+  '서울시 강북구',
+  '서울시 도봉구',
+  '서울시 노원구',
+  '서울시 은평구',
+  '서울시 서대문구',
+  '서울시 마포구',
+  '서울시 양천구',
+  '서울시 강서구',
+  '서울시 구로구',
+  '서울시 금천구',
+  '서울시 영등포구',
+  '서울시 동작구',
+  '서울시 관악구',
+  '서울시 서초구',
+  '서울시 강남구',
+  '서울시 송파구',
+  '서울시 강동구',
+];
+// Todo: key값 수정, 적용하기 버튼 onClick 이벤트 추가하기, ESlint 들여쓰기 이슈 고치기
+function DetailFilterModal() {
+  const [activeModal, setActiveModal] = useState<boolean>(false);
+  const [startDate, setStartDate] = useState<Date | null>(null);
+  const [clickAddressList, setClickAddressList] = useState<string[]>([]);
+  const [filterAddressList, setFilterAddressList] = useState<string[]>([]);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [amount, setAmount] = useState<number | null>();
+  const handleAddClickAddressListItem = (address: string) => {
+    if (clickAddressList.indexOf(address) !== -1) return;
+    const addressAddArr = [...clickAddressList, address];
+    setClickAddressList(addressAddArr);
+  };
+  const handleDeleteClickAddressListItem = (address: string) => {
+    const sliceArr = [...clickAddressList];
+    const addressIndex = clickAddressList.indexOf(address);
+    sliceArr.splice(addressIndex, 1);
+    setClickAddressList(sliceArr);
+  };
```

Comment:
P4: `splice`가 원본 배열을 변경하기 때문에 복사본을 만드는 방식으로 작성해주신 디테일 너무 좋네요👍 개인적인 의견을 말씀드리자면, 이 부분을 `filter`를 써서 만들어보는 것도 괜찮을 것 같습니다...!

```suggestion
  const handleDeleteClickAddressListItem = useCallback((address: string) => {
  setClickedAddresses(prev => prev.filter(item => item !== address));
}, []);
```